benchmark/benchmark_adapter.cpp

```diff
@@ -81,8 +81,13 @@ namespace xt
             auto ab = xt::adapt(b);
             auto ar = xt::adapt(res);
             auto fun = aa + ab;
-            std::transform(fun.linear_cbegin(), fun.linear_cend(), ar.linear_begin(),
-                           [](typename decltype(fun)::value_type x) { return static_cast<typename decltype(ar)::value_type>(x); });
+            std::transform(
+                fun.linear_cbegin(),
+                fun.linear_cend(),
+                ar.linear_begin(),
+                [](typename decltype(fun)::value_type x)
+                { return static_cast<typename decltype(ar)::value_type>(x); }
```

Comment:
I agree, however there are no options to get this right.
We can
- A: Always break lambdas, so with a simpler one it would look like
  ```cpp
  std::transform(
      ...,
      [](auto x)
      {
          return x;
      }
  );
  ```
- B: Keep like this, but re-write the lambda so it's shorter (e.g. use `auto`), or longer (e.g. typdef the type of `static_cast`) since two lines lambdas will format well.
- C: Format manually and surround with `# clang-format on/off`
- D: live with it

Reply:
`A` looks like a good compromise, I will try this formatting in real code to see if I still have the same impression of more arguments than actually passed.
`B` cannot guarantee the formatting in all cases, and using `auto` as an argument moves the type check into the lambda body instead of having it at the site call.
`C` good to know that `clang-format` can be disabled, however we should keep this option for very uncommon syntactic constructions
`D`: no way, I will never surrender ;)